backend/src/equipment/equipment.service.ts

```diff
@@ -8,16 +12,33 @@ import {
   NotFoundError,
   PrismaClientKnownRequestError,
 } from '@prisma/client/runtime';
+import { PostgresErrorCode } from '../shared/constants/postgress-error-codes.enum';
 
 @Injectable()
 export class EquipmentService {
   constructor(private readonly prisma: PrismaService) {}
 
   public async addEquipment(data: EquipmentDTO): Promise<EquipmentDTO> {
-    const equipment = await this.prisma.equipment.create({
-      data,
-    });
-    return EquipmentService.convertToDTO(equipment);
+    try {
+      const equipment = await this.prisma.equipment.create({
+        data,
+      });
+      return EquipmentService.convertToDTO(equipment);
+    } catch (error) {
+      if (error?.code === PostgresErrorCode.UniqueViolation) {
+        if (error.meta.target[0] === 'serial') {
```

Comment:
used the `meta.target` property of the thrown `error` to identify what field unique field in the equipment is existing